File: backend/api/rag/v1/chunks.go
package v1

import (
	"backend/internal/model/entity"

	"github.com/gogf/gf/v2/frame/g"
)

type ChunksListReq struct {
	g.Meta         `path:"/v1/chunksList" method:"get" tags:"rag" sm:"获取文档片段列表"`
	KnowledgeDocId int64 `p:"knowledge_doc_id" dc:"knowledge_doc_id" v:"required"`
	Page           int   `p:"page" dc:"page" v:"required|min:1" d:"1"`
	Size           int   `p:"size" dc:"size" v:"required|min:1|max:100" d:"10"`
}

type ChunksListRes struct {
	g.Meta `mime:"application/json"`
	Data   []*entity.KnowledgeChunks `json:"data"`
	Total  int                       `json:"total"`
	Page   int                       `json:"page"`
	Size   int                       `json:"size"`
}

type ChunkDeleteReq struct {
	g.Meta `path:"/v1/chunksDelete" method:"delete" tags:"rag" sm:"删除文档片段"`
	Id     int64 `p:"id" dc:"id" v:"required"`
}

type ChunkDeleteRes struct {
	g.Meta `mime:"application/json"`
}

type UpdateChunkReq struct {
	g.Meta `path:"/v1/chunksPut" method:"put" tags:"rag" sm:"更新文档片段状态"`
	Ids    []int64 `p:"ids" dc:"ids" v:"required"`
	Status int     `p:"status" dc:"status" v:"required|in:0,1"`
}

type UpdateChunkRes struct {
	g.Meta `mime:"application/json"`
}

type UpdateChunkContentReq struct {
	g.Meta  `path:"/v1/chunks_content" method:"put" tags:"rag" sm:"更新文档片段内容"`
	Id      int64  `p:"id" dc:"id" v:"required"`
	Content string `p:"content" dc:"content" v:"required"`
}

type UpdateChunkContentRes struct {
	g.Meta `mime:"application/json"`
}

File: frontChat/src/services/chunks.ts
import { ApiClient } from '../utils/axios';

// 知识块状态枚举
export enum ChunkStatus {
  ACTIVE = 0,    // 启用
  DISABLED = 1   // 禁用
}

// 知识块数据类型
export interface KnowledgeChunk {
  id: number;
  knowledgeDocId: number;
  chunkId: string;
  content: string;
  ext: string;
  status: ChunkStatus;
  createdAt: string;
  updatedAt: string;
}

// API 请求/响应类型
export interface ChunksListReq {
  knowledge_doc_id: number;
  page?: number;
  size?: number;
}

export interface ChunksListRes {
  data: KnowledgeChunk[];
  total: number;
  page: number;
  size: number;
}

export interface ChunkDeleteReq {
  id: number;
}

export interface UpdateChunkReq {
  ids: number[];
  status: ChunkStatus;
}

export interface UpdateChunkContentReq {
  id: number;
  content: string;
}

const BASE_PATH = '/gateway/v1';

/**
 * 知识块 API 服务
 */
export const ChunksService = {
  /**
   * 获取知识块列表
   */
  getList: async (params: ChunksListReq): Promise<ChunksListRes> => {
    const queryParams = {
      knowledge_doc_id: params.knowledge_doc_id,
      page: params.page || 1,
      size: params.size || 10
    };
    return ApiClient.get<ChunksListRes>(`${BASE_PATH}/chunksList`, queryParams);
  },

  /**
   * 删除知识块
   */
  delete: async (params: ChunkDeleteReq): Promise<void> => {
    return ApiClient.delete<void>(`${BASE_PATH}/chunksDelete?id=${params.id}`);
  },

  /**
   * 批量更新知识块状态
   */
  updateStatus: async (data: UpdateChunkReq): Promise<void> => {
    return ApiClient.put<void>(`${BASE_PATH}/chunksPut`, data);
  },

  /**
   * 更新知识块内容
   */
  updateContent: async (data: UpdateChunkContentReq): Promise<void> => {
    return ApiClient.put<void>(`${BASE_PATH}/chunks_content`, data);
  }
};

export default ChunksService;
